nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-parameter-contexts.js

```diff
@@ -0,0 +1,2298 @@
+                                                    {
+                                                        content: list
+                                                    }));
+                                            }
+                                        });
+                                    });
+                                });
+                            }
+                        } else {
+                            groupTwist.find('.referencing-components-template').remove();
+                        }
+
+                        // toggle this block
+                        toggle(twist, list);
+
+                        // update the border if necessary
+                        updateReferencingComponentsBorder($('#parameter-referencing-components-container'));
+                    }).append(twist).append(title).append(count).appendTo(referencingPgReferencesContainer);
+
+                    // add the listing
+                    list.appendTo(referencingPgReferencesContainer);
+
+                    // expand the group twist
+                    groupTwist.click();
+
+                    // update the border if necessary
+                    updateReferencingComponentsBorder($('#parameter-referencing-components-container'));
+                };
+
+                // get process group and show name or UUID based on user permissions
+                var pg = nfProcessGroup.get(affectedProcessGroup.id);
+                if (!nfCommon.isUndefinedOrNull(pg)) {
+                    if (pg.permissions.canRead === true) {
+                        // create block for this process group
+                        createReferenceBlock(pg.component.name, groups);
+                    } else {
+                        // create block for this process group
+                        createReferenceBlock(affectedProcessGroup.id, groups);
+                    }
+                } else {
+                    var processGroupName = affectedProcessGroup.id;
+
+                    // attempt to resolve the group name
+                    var breadcrumbs = nfNgBridge.injector.get('breadcrumbsCtrl').getBreadcrumbs();
+                    $.each(breadcrumbs, function (_, breadcrumbEntity) {
+                        if (breadcrumbEntity.id === affectedProcessGroup.id) {
+                            processGroupName = breadcrumbEntity.label;
+                            return false;
+                        }
+                    });
+                    // create block for this process group
+                    createReferenceBlock(processGroupName, groups);
+                }
+            });
+        }
+    };
+
+    /**
+     * Sorts the specified entities based on the name.
+     *
+     * @param {object} a
+     * @param {object} b
+     * @returns {number}
+     */
+    var nameComparator = function (a, b) {
+        return a.component.name.localeCompare(b.component.name);
+    };
+
+    var parameterKeyRegex = /^[a-zA-Z0-9-_. ]+/;
```

Comment:
this needs to be
```var parameterKeyRegex = /^[a-zA-Z0-9-_. ]+$/;```

otherwise it is only enforcing that parameters start with valid characters.